addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/AlertFilter.java

```diff
@@ -146,11 +241,22 @@ public static String encode(AlertFilter alertFilter) {
             out.append(Base64.encodeBase64String(alertFilter.url.getBytes()));
         }
         out.append(FIELD_SEPARATOR);
-        out.append(alertFilter.isRegex()).append(FIELD_SEPARATOR);
+        out.append(alertFilter.isUrlRegex()).append(FIELD_SEPARATOR);
         if (alertFilter.parameter != null) {
             out.append(Base64.encodeBase64String(alertFilter.parameter.getBytes()));
         }
         out.append(FIELD_SEPARATOR);
+        out.append(alertFilter.isParameterRegex()).append(FIELD_SEPARATOR);
+        if (alertFilter.attack != null) {
+            out.append(Base64.encodeBase64String(alertFilter.attack.getBytes()));
```

Comment:
These byte/string conversions are relying on the platform's default charset, this can cause differences/issues when moving the session/context around (e.g. Windows vs Linux). Fine if handled in a follow up PR, existing fields already have that problem.